crates/core/src/test_utils/execution_helpers.rs

```diff
@@ -1,10 +1,35 @@
+use std::sync::Arc;
+
 use ethers::signers::{LocalWallet, Signer};
 use reth_primitives::{Address, BlockId, H256, U256};
+use starknet::accounts::{Account, Call, ConnectedAccount, SingleOwnerAccount};
+use starknet::core::types::InvokeTransactionResult;
+use starknet::providers::jsonrpc::HttpTransport;
+use starknet::providers::JsonRpcClient;
+use starknet::signers::LocalWallet as StarknetLocalWallet;
 
 use super::deploy_helpers::{create_eth_transfer_tx, create_raw_ethereum_tx, KakarotTestEnvironmentContext};
 use crate::client::api::KakarotEthApi;
+use crate::client::waiter::TransactionWaiter;
+
+pub async fn execute_and_wait_for_tx(
+    account: &SingleOwnerAccount<JsonRpcClient<HttpTransport>, StarknetLocalWallet>,
+    calls: Vec<Call>,
+) -> InvokeTransactionResult {
+    let c = calls.clone();
```

Comment:
not sure about this but instinctively, if the signature of execute is to take in a Vec of calls,
Then why not let this function take ownership of calls, instead of cloning it?

Reply:
So the issue is that in case of a panic, I want to log the call vector that caused the failure. However, account.execute moves the value of calls, since it takes a Vec<Call>. So I need to clone it in order to keep the value and print in the unwrap_or_else.

Reply:
wouldn't `let res = account.execute(calls.clone()).send().await.expect(&format!("Failed to execute tx: {:?}", calls));` be more concise?

Reply:
sure can update to this.